app/src/ui/lib/list/list-row.tsx

```diff
@@ -27,48 +27,59 @@ interface IListRowProps {
   readonly onRef?: (element: HTMLDivElement | null) => void
 
   /** callback to fire when the row receives a mouseover event */
-  readonly onRowMouseOver: (index: number, e: React.MouseEvent<any>) => void
+  readonly onRowMouseOver?: (index: number, e: React.MouseEvent<any>) => void
```

Comment:
These fields were made optional to simply creating a `ListRow` outside of the `List` component.

Reply:
> to simply creating a `ListRow` outside of the `List`

I'd love some extra context on why we're using `ListRow` outside of `List` here and what we gain, as hiding these callbacks stands out as something worth unpacking.

I'd also love some @desktop/design input on whether there's a better way we can do this - could we borrow the stylings from and create a standalone component that just wraps a `div`?

Reply:
The ListRow component gives me a few things for free:
* Styling - the _Stashed Changes_ item in the design looks like a list item
* Interaction - the _Stashed Changes_  item is selectable and comes with hooks to respond to keyboard and mouse events

I'm fine with coming up with a new component in an iteration however. 👍

Reply:
> Styling - the Stashed Changes item in the design looks like a list item

I see you're already adding some custom styles to `_changes-list` as part of this PR - what if we tried using a mixin here that allowed for reuse of the `list-item` styles in different areas? Would that get you most of the way to something usable here?

https://github.com/desktop/desktop/blob/4e1ad40f7ed935dfa56c1f8208b8054e61bd380f/app/styles/ui/_list.scss#L76-L100

> Interaction - the Stashed Changes item is selectable and comes with hooks to respond to keyboard and mouse events

This is fair, but I think you could crimp the selection logic and click events from inside `ListRow` without needing to change the behaviour of `ListRow`. Looking at the setup, there's a lot of hard-coded values because of what `ListRow` is expecting:

```jsx
<ListRow
          rowCount={1}                             // necessary?
          rowIndex={0}                              // necessary?
          selectable={true}                       // necessary?
          selected={this.props.isShowingStashEntry} 
          onRowClick={this.onStashEntryClicked}
          tabIndex={0}                               // necessary?
          style={StashListRowStyle}        // pass in rowHeight directly?
          className="stash-entry-row" // could live inside new component
        >
```

I'd love to see an exploration of alternatives here because I think you could get away with something simpler than `ListRow` for your use case.

Reply:
Resolving this because #7263 will address this (and is currently under review)